egui_demo_lib/src/easy_mark/easy_mark_editor.rs

```diff
@@ -117,59 +117,30 @@ impl EasyMarkEditor {
 
 fn shortcuts(ui: &Ui, code: &mut dyn TextBuffer, ccursor_range: &mut CCursorRange) -> bool {
     let mut any_change = false;
-    for event in &ui.input().events {
-        if let Event::Key {
+    for (key, surrounding) in [
+        (Key::B, "*"), // *bold*
+        (Key::C, "`"), // `code`
+        (Key::I, "/"), // /italics/
+        (Key::L, "$"), // $subscript$
+        (Key::R, "^"), // ^superscript^
+        (Key::S, "~"), // ~strikethrough~
+        (Key::U, "_"), // _underline_
+    ] {
+        if ui.input_mut().consume_key(
             key,
-            pressed: true,
-            modifiers,
-        } = event
-        {
-            if modifiers.command_only() {
-                match &key {
-                    // toggle *bold*
-                    Key::B => {
-                        toggle_surrounding(code, ccursor_range, "*");
-                        any_change = true;
-                    }
-                    // toggle `code`
-                    Key::C => {
-                        toggle_surrounding(code, ccursor_range, "`");
-                        any_change = true;
-                    }
-                    // toggle /italics/
-                    Key::I => {
-                        toggle_surrounding(code, ccursor_range, "/");
-                        any_change = true;
-                    }
-                    // toggle $lowered$
-                    Key::L => {
-                        toggle_surrounding(code, ccursor_range, "$");
-                        any_change = true;
-                    }
-                    // toggle ^raised^
-                    Key::R => {
-                        toggle_surrounding(code, ccursor_range, "^");
-                        any_change = true;
-                    }
-                    // toggle ~strikethrough~
-                    Key::S => {
-                        toggle_surrounding(code, ccursor_range, "~");
-                        any_change = true;
-                    }
-                    // toggle _underline_
-                    Key::U => {
-                        toggle_surrounding(code, ccursor_range, "_");
-                        any_change = true;
-                    }
-                    _ => {}
-                }
-            }
-        }
+            egui::Modifiers {
+                command: true,
+                ..Default::default()
```

Comment:
We can make this even nicer by switching the argument order and adding some helpers to `Modifiers`:

```rust
            if ui.input_mut().consume_key(egui::Modifiers::command(), key) {
```

Reply:
nice, i added it was a builder api but lmk if you prefer classmethods + operator overloading instead